docs/volumes.md

```diff
@@ -0,0 +1,96 @@
+# Working with Volumes
+
+The below are the volumes that can be mounted for XL products and the reason to use them
+
+## XL Deploy
+
+The env variables `APP_ROOT` ponts to `/opt/xebialabs` and `APP_HOME` point to  `/${APP_ROOT}/xl-deploy-server` by default, there is no need to change them usually.
+
+The following locations can be mounted as volumes to provide either configuration data, and/or persistent storage for application data.
+
+**${APP_HOME}/conf** - This directory contains the configuration files and keystores for XL Deploy. If mounted it is possible to configure (and inject) non-default configuration settings. Some configurations are always overwritten by the container startup script based on env variables set, to avoid that you can set the `GENERATE_XL_CONFIG` env variable to `false`
+
+**${APP_HOME}/hotfix/lib** - This directory contains the hotfixes for the libraries used by XL Deploy. When instructed by XebiaLabs support personnel, you should drop any delivered hotfixes in here.
+
+**${APP_HOME}/hotfix/plugins** - This directory contains the hotfixes for the plugins used by XL Deploy. When instructed by XebiaLabs support personnel, you should drop any delivered plugin hotfixes in here.
+
+**${APP_HOME}/ext** - This directory contains the developed (exploded plugins) for XL Deploy. You can customize an existing plugin here, for example you can modify the `synthetic.xml` for a specific plugin here.
+
+**${APP_HOME}/plugins** - This directory contains the plugins that are running in XL Deploy. When mounted only plugins present in this directory will be loaded and you need to ensure to provide all required base plugins here. If you mount an empty directory the default bundled plugins will be loaded.
+
+**${APP_HOME}/repository** - This directory contains the embedded repository database for XL Deploy. Using the configuration injected through the ${APP_HOME}/conf volume, it is possible to configure the used database to a remotely running database engine, instead of using the embedded memory based one.
```

Comment:
we should discourage users using local db and repository folder. this is only for testing only

Reply:
Ok i'll add note